### ISSUE TITLE
Can it be designed to judge whether the point cloud is perpendicular to a certain plane

### ISSUE THREAD

**Issue by Longyouspongecake** [rubenwiersma/deltaconv#1]
This work is very interesting, I think if I have a group of point clouds with different shapes but all perpendicular to the xy plane (similar to walls with various shapes perpendicular to the ground), can I use your algorithm directly to determine how much they are perpendicular to the ground (some walls are not exactly perpendicular to the ground)

**Comment by rubenwiersma** [comment]
Thank you, @Longyouspongecake! I think a neural network with DeltaConv could be fitted in some way to help with your question. However, it might be helpful to first answer the question if a data-driven approach is really necessary for your task. One way to answer that question is to see if a simpler (non-learning) approach works. If it doesn't, can you expect the 'missing information' to be learned from data?

An example of a non-learning approach is to use PCA to find the principal components of the point cloud or a region of the point cloud. The eigenvector that points into the z-direction will give you a clue about how perpendicular your point cloud is to the ground. For an example of how to compute PCA on a point cloud to find principal axes, you could look at https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/transforms/normalize_rotation.py

The vector you would check is the column in `v` that has the largest z-value in this line:
https://github.com/pyg-team/pytorch_geometric/blob/9a8ae988a79b38b98e11924fbcd40871c174d06c/torch_geometric/transforms/normalize_rotation.py#L36
